### ISSUE TITLE
twisted.internet.endpoints._parseSSL defaults to SSLv23_METHOD, should default to None

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7684]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#7684|
|Type|defect|
|Created|2014-10-16 18:17:21Z|
|Branch|https://github.com/twisted/twisted/tree/ssl-version-default-7684|

If you specify `SSLv23_METHOD`, `CertificateOptions` only defaults to disabling SSLv2.  However, if you specify `None`, i.e. what `CertificateOptions` deems to be the default, it will disable SSLv3.

Attachments:

* [t7684.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c6f/c6f73c625e8d1f4ef87845750ba0f46a5e67fd3f/5bca5e56357ae2c08ee7d7171388378a9d43560c.diff) (2067 bytes) - added by Alex on 2014-10-16 18:23:15Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7684 7684
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_ssl_version_default_7684 branches/ssl-version-default-7684
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1413483441747109 1413483441747109
changetime__1413488637578360 1413488637578360
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Alex set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

(In [43391]) Branching to ssl-version-default-7684.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

Looks like the buildbot is happy.  The change looks good.  I am going to apply exarkun's suggested NEWS file change and land.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

(In [43395]) Merge ssl-version-default-7684: Disable SSLv3 by default for string server endpoints.

Author: Alex

Reviewer: glyph

Fixes: [#7684](https://github.com/twisted/twisted/issues/7684)

...
